### PR TITLE
cpu/esp32: use CPU_FAM and CPU_ARCH for ESP32x SoC variant dependent compilation

### PR DESCRIPTION
**Pull request by gschorcht**
### Contribution description

This PR provides the changes that are necessary to compile the ESP32 port for different ESP32x  SoC variants (families). It includes the following changes: 
- `CPU_FAM` is used instead of `CPU` in Makefiles
- `CPU_ARCH` is used instead of `CPU` in Makefiles to define architecture dependent compile options
- `CPU_FAM_*` is used instead of `MCU_*` in source code
- `CPU_FAM` definition is moved to board definition
-  `CPU_*` definitions are moved to CPU-specific `Makefile.features`

The PR could be splitted into 3 separate smaller PRs if necessary.

### Testing procedure

Green CI

### CODE REVIEW

File: cpu/esp32/Makefile.include [hunk 1]
```diff
@@ -1,16 +1,17 @@
 # ESP32x specific flashing options
-FLASH_CHIP  = $(CPU)
-ifneq (,$(filter esp32,$(CPU)))
+FLASH_CHIP  = $(CPU_FAM)
+ifneq (,$(filter esp32,$(CPU_FAM)))
```
Comment (benpicco) [inline cpu/esp32/Makefile.include:3]:
```suggestion
ifeq (esp32,$(CPU_FAM))
```

`CPU_FAM` is not a list

Comment (gschorcht) [reply cpu/esp32/Makefile.include:3]: Yes, I will change it for the moment. However, in future PRs for ESP32-S2 and ESP32-S3 were multiple families will match, we will need the `filter` function.

Comment (benpicco) [reply cpu/esp32/Makefile.include]: Ah true, if this gets expanded in the future filter is warranted - anyway feel free to squash

**Review by benpicco — APPROVED** [review]
If Murdock is happy, so am I

**Pull request merged**